File: ModuPOS.Api/Entities/Producto.cs
namespace ModuPOS.Api.Entities
{
    public class Producto : BaseEntity
    {
        public string SKU { get; set; } = string.Empty;
        public string Nombre { get; set; } = string.Empty;
        public decimal PrecioActual { get; set; }
        public int Stock { get; set; }

        public int? CategoriaId { get; set; }
        public virtual Categoria? Categoria { get; set; }

        public int? ImagenId { get; set; }
        public virtual Imagen? Imagen { get; set; }
    }
}

File: backend/src/main/java/com/jse/modupos/model/Producto.java
package com.jse.modupos.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "productos")
public class Producto extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(unique = true, length = 50)
    private String sku;

    @Column(name = "precio_actual", nullable = false, precision = 18, scale = 4)
    private BigDecimal precioActual;

    @Column(nullable = false)
    private Integer stock = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id", nullable = false)
    private Categoria categoria;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "imagen_id", referencedColumnName = "id")
    private Imagen imagen;
}
